apps/ucp-ui/src/Institutions/Institutions.test.tsx

```diff
@@ -303,4 +308,95 @@ describe("<Institutions />", () => {
       await screen.findByText(INSTITUTIONS_EMPTY_RESULTS_TEXT),
     ).toBeInTheDocument();
   });
+
+  it("initially shows no arrows, then shows down arrow when the first column is clicked, and then up arrow when it's clicked again", async () => {
+    const { container } = render(<Institutions />);
+
+    await waitForLoad();
+
+    expect(
+      container.querySelector(
+        `${INSTITUTIONS_TABLE_ROW_ACTIVE_CLASS} ${INSTITUTIONS_TABLE_SORT_ARROW_CLASS_UP}`,
+      ),
+    ).not.toBeInTheDocument();
+
+    expect(
+      container.querySelector(
+        `${INSTITUTIONS_TABLE_ROW_ACTIVE_CLASS} ${INSTITUTIONS_TABLE_SORT_ARROW_CLASS_DOWN}`,
+      ),
+    ).not.toBeInTheDocument();
+
+    await userEvent.click(
+      screen.getByText(INSTITUTIONS_TABLE_INSTITUTION_HEADER_TITLE),
+    );
+
+    expect(
+      container.querySelector(INSTITUTIONS_TABLE_SORT_ARROW_CLASS_DOWN),
+    ).toBeInTheDocument();
+
+    await userEvent.click(
+      screen.getByText(INSTITUTIONS_TABLE_INSTITUTION_HEADER_TITLE),
+    );
+
+    expect(
+      container.querySelector(INSTITUTIONS_TABLE_SORT_ARROW_CLASS_UP),
+    ).toBeInTheDocument();
+  });
+
+  it("sorts correctly when headers are clicked", async () => {
+    const testSortList = [
+      {
+        ...testInstitution,
+        id: "9e7eaf0f-f3dc-4bd8-97b3-e09d85dcd0fa",
+        name: "AA Bank",
+      },
+      {
+        ...testInstitution,
+        id: "a47eca28-8929-410f-8f7e-317c9678110c",
+        name: "ZZ Bank",
+      },
+    ];
+
+    const { container } = render(<Institutions />);
+
+    await waitForLoad();
+
+    server.use(
+      http.get(INSTITUTION_SERVICE_INSTITUTIONS_URL, () =>
+        HttpResponse.json({
+          ...institutionsPage1,
+          institutions: [...testSortList],
+        }),
+      ),
+    );
+
+    await userEvent.click(
+      screen.getByText(INSTITUTIONS_TABLE_INSTITUTION_HEADER_TITLE),
+    );
+
+    await waitForLoad();
+
+    expect(
+      container?.querySelector(INSTITUTIONS_TABLE_ROW_ROOT_CLASS)?.children[1], // Second column
+    ).toHaveTextContent(testSortList[0].id);
+
+    server.use(
```

Comment:
This tells us that it's calling the server again and looking for new results, but it doesn't really tell us that the sort parameters are integrated correctly.

I would setup the handler one time above, check the query parameters for sorting, and return the institutions with a different sort based on the value you read from the query parameters.